### PR TITLE
149 reduce docker images size

### PR DESCRIPTION
**Pull request by Pbarbosa92**
This PR aims to decrease the size of each simulator docker image.

Issue https://github.com/inductiva/tasks/issues/149

### CODE REVIEW

File: simulators/gromacs/v2022.2/Dockerfile [hunk 1]
```diff
@@ -11,10 +10,56 @@ RUN apt-get update && apt-get install -y \
...
 RUN chmod +x /home/test_sim.sh
 
-# Start from a clen GROMACS image
-FROM gromacs/gromacs:2022.2
+From ubuntu:22.04 as build
 
-# Set environment variable to allow GPU direct communication
-ENV GMX_ENABLE_DIRECT_GPU_COMM=1
+ENV OMPI_ALLOW_RUN_AS_ROOT=1
+ENV OMPI_ALLOW_RUN_AS_ROOT_CONFIRM=1
```
Comment (IvanPombo) [inline simulators/gromacs/v2022.2/Dockerfile:16]: Not sure if this should exist here, it is not adequate to allow run as root in HPC, since it allows root from inside the container to do whatever it wants. In principle, inside the container users can do `--allow-run-as-root` and hence at least they are more aware of it.

In any case, we should start considering applying `Users` inside the dockerfiles, wdyt @sergiomsantos ?

File: simulators/gromacs/v2022.2/Dockerfile [hunk 2]
```diff
@@ -11,10 +10,56 @@ RUN apt-get update && apt-get install -y \
...
+    #check can fail if the computer is slow or has few resources
+    #make check && \
+    make install && \
+    . /usr/local/gromacs/bin/GMXRC && \
+    mv /usr/local/gromacs/bin/gmx_mpi /usr/local/gromacs/bin/gmx && \
+    rm -r /gromacs-2022.2
+
+From ubuntu:22.04 
+
+Copy --from=build /usr/local/gromacs /usr/local/gromacs
```
Comment (IvanPombo) [inline simulators/gromacs/v2022.2/Dockerfile:50]: Does this work?

File: simulators/gromacs/v2022.2/Dockerfile [hunk 3]
```diff
@@ -11,10 +10,56 @@ RUN apt-get update && apt-get install -y \
...
+
+ENV PATH="${PATH}:/usr/local/gromacs/bin/"
+    
+Copy --from=test_env /home /home
```
Comment (IvanPombo) [inline simulators/gromacs/v2022.2/Dockerfile:54]: Does this work?

File: simulators/gromacs/v2022.2/Dockerfile [hunk 4]
```diff
@@ -11,10 +10,56 @@ RUN apt-get update && apt-get install -y \
...
+
+Run apt-get update -y && \
```
Comment (IvanPombo) [inline simulators/gromacs/v2022.2/Dockerfile:56]: Does this work?

File: simulators/gromacs/v2022.2/Dockerfile [hunk 5]
```diff
@@ -11,10 +10,56 @@ RUN apt-get update && apt-get install -y \
...
+    apt-get install -y openmpi-bin
+
+Copy --from=build /usr/lib/x86_64-linux-gnu/libevent_pthreads-2.1.so.7 /usr/lib/x86_64-linux-gnu/libevent_pthreads-2.1.so.7
+Copy --from=build /usr/lib/x86_64-linux-gnu/libevent_core-2.1.so.7 /usr/lib/x86_64-linux-gnu/libevent_core-2.1.so.7
+Copy --from=build /usr/lib/x86_64-linux-gnu/libopen-rte.so.40 /usr/lib/x86_64-linux-gnu/libopen-rte.so.40
+Copy --from=build /usr/lib/x86_64-linux-gnu/libopen-pal.so.40 /usr/lib/x86_64-linux-gnu/libopen-pal.so.40
+Copy --from=build /usr/lib/x86_64-linux-gnu/libhwloc.so.15 /usr/lib/x86_64-linux-gnu/libhwloc.so.15
+Copy --from=build /usr/lib/x86_64-linux-gnu/libgomp.so.1 /usr/lib/x86_64-linux-gnu/libgomp.so.1
+Copy --from=build /usr/lib/x86_64-linux-gnu/libmpi.so.40 /usr/lib/x86_64-linux-gnu/libmpi.so.40
```
Comment (IvanPombo) [inline simulators/gromacs/v2022.2/Dockerfile:65]: Does this work?

File: simulators/openfast/v3.5.2/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    unzip /home/openfast-input-example.zip -d /home/ && \
+    rm /home/openfast-input-example.zip && \
+    ls -l
+
+COPY ./test_sim.sh /home/test_sim.sh
+RUN chmod +x /home/test_sim.sh
+
+FROM ubuntu:22.04 as build
+
+ENV DEBIAN_FRONTEND=noninteractive TZ=America/Denver
```
Comment (IvanPombo) [inline simulators/openfast/v3.5.2/Dockerfile:16]: Why is the timezone required?

File: simulators/openfast/v3.5.2/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+Copy --from=test_env /home /home
+
+Copy --from=build /usr/lib/x86_64-linux-gnu/libquadmath.so.0 /usr/lib/x86_64-linux-gnu/libquadmath.so.0
+Copy --from=build /usr/lib/x86_64-linux-gnu/libgfortran.so.5 /usr/lib/x86_64-linux-gnu/libgfortran.so.5
+Copy --from=build /usr/lib/x86_64-linux-gnu/liblapack.so.3 /usr/lib/x86_64-linux-gnu/liblapack.so.3
+Copy --from=build /usr/lib/x86_64-linux-gnu/libblas.so.3 /usr/lib/x86_64-linux-gnu/libblas.so.3
+Copy --from=build /usr/lib/x86_64-linux-gnu/libgomp.so.1 /usr/lib/x86_64-linux-gnu/libgomp.so.1
+
+# Copy the binaries and shared libraries to the appropriate directories
+Copy --from=build /openfast/install/bin/* /usr/local/bin/
```
Comment (IvanPombo) [inline simulators/openfast/v3.5.2/Dockerfile:81]: Does this work?

File: simulators/splishsplash/v2.13.0/Dockerfile [hunk 8]
```diff
@@ -70,4 +52,23 @@ RUN make -j 4
...
+Copy --from=build /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 /usr/lib/x86_64-linux-gnu/libXdmcp.so.6
+Copy --from=build /usr/lib/x86_64-linux-gnu/libgomp.so.1 /usr/lib/x86_64-linux-gnu/libgomp.so.1
+Copy --from=build /usr/lib/x86_64-linux-gnu/libX11.so.6 /usr/lib/x86_64-linux-gnu/libX11.so.6
+Copy --from=build /usr/lib/x86_64-linux-gnu/libGLU.so.1 /usr/lib/x86_64-linux-gnu/libGLU.so.1
+Copy --from=build /usr/lib/x86_64-linux-gnu/libxcb.so.1 /usr/lib/x86_64-linux-gnu/libxcb.so.1
+Copy --from=build /usr/lib/x86_64-linux-gnu/libXau.so.6 /usr/lib/x86_64-linux-gnu/libXau.so.6
+Copy --from=build /usr/lib/x86_64-linux-gnu/libbsd.so.0 /usr/lib/x86_64-linux-gnu/libbsd.so.0
+Copy --from=build /usr/lib/x86_64-linux-gnu/libmd.so.0 /usr/lib/x86_64-linux-gnu/libmd.so.0
+Copy --from=build /usr/lib/x86_64-linux-gnu/libX11.so /usr/lib/x86_64-linux-gnu/libX11.so
+Copy --from=build /usr/lib/x86_64-linux-gnu/libX11.a /usr/lib/x86_64-linux-gnu/libX11.a
```
Comment (IvanPombo) [inline simulators/splishsplash/v2.13.0/Dockerfile:71]: Does this work?

File: simulators/openfast/v3.5.2/Dockerfile
Comment (IvanPombo) [inline simulators/openfast/v3.5.2/Dockerfile:1]: This should be in a separate PR, since it is adding a new simulator and not reducing it's size.

File: simulators/openfast/v3.5.2/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+RUN chmod +x /tmp/Miniconda3-py312_24.1.2-0-Linux-x86_64.sh && \
+    bash /tmp/Miniconda3-py312_24.1.2-0-Linux-x86_64.sh -b && \
+    conda config --add channels conda-forge && \
+    conda create -y --name rosco-env python=3.10 && \
+    conda init bash && \
+    . ~/.bashrc && \
+    conda activate rosco-env && \
+    git clone https://github.com/NREL/ROSCO.git && \
+    cd ROSCO && \
+    pip install -e .
```
Comment (IvanPombo) [inline simulators/openfast/v3.5.2/Dockerfile:64]: Just curious, why do we need miniconda? Even if we need to install any Python package we could directly do it with Python/Pip as numpy above, no? Why is an environment required?

File: simulators/gromacs/v2022.2/Dockerfile (hunk 1 above)
Comment (Pbarbosa92) [reply simulators/gromacs/v2022.2/Dockerfile:16]: If I don't use this ENV var the frolics test that use mpi fail because we are running them as root.

File: simulators/dualsphysics/v5.2.1/Dockerfile [hunk 10]
```diff
@@ -25,28 +25,56 @@ RUN apt-get update -y \
...
+
+# Source: https://github.com/DualSPHysics/DualSPHysics/wiki/5.-Running-DualSPHysics
+# Folder src:
+# The folder source contains the source files of v5.0 (.cpp, .cu and .h). The makefile of linux and the text file for CMAKE.txt are also included in this folder.
+# Visual studio (Community 2015) project for windows can be found in VS.
+# The libraries (.a and .lib) necessary for compilation are included in lib.
+# Folder src_extra:
+# The release includes not only the source files of DualSPHysics v5.0 but also the source files of a code named “ToVTK5”. This code is provided to show how to load particle data, how to read .bi4 files and how to create .vtk or .csv files.
+# Folder src_mphase:
+# The source files of the of DualSPHysics_v4.0_LiquidGas and DSPH_v5.0_NNmphase are included here.
```
[the remark this answers is not in the document]
Comment (Pbarbosa92) [reply simulators/dualsphysics/v5.2.1/Dockerfile:45]: Removed the comment. I rather have no information than have misleading or wrong information.

File: simulators/openfast/v3.5.2/Dockerfile
Comment (Pbarbosa92) [reply simulators/openfast/v3.5.2/Dockerfile:1]: Removed it from here. Will add it back in another PR

File: simulators/openfast/v3.5.2/Dockerfile (hunk 9 above)
Comment (Pbarbosa92) [reply simulators/openfast/v3.5.2/Dockerfile]: You are right. We can build the files we need without miniconda.

File: simulators/gromacs/v2022.2/Dockerfile (hunk 1 above)
Comment (IvanPombo) [reply simulators/gromacs/v2022.2/Dockerfile:16]: Yes, I know, but the reason is to use `--allow-run-as-root` which implies that you are actually overriding it.

**Review by IvanPombo — APPROVED** [review]
LGTM!
